### PR TITLE
Add makefiles example

### PR DESCRIPTION
**Pull request by danimtb**
From https://github.com/memsharded/conan-example-makefiles in https://docs.conan.io/en/latest/howtos/makefiles.html

### CODE REVIEW

**Comment by uilianries** [comment]
Great idea! 👏

**Comment by danimtb** [comment]
Well, we don't have `make` in Appveyor so the example is failing there. Not sure if there is a solution for this or we should just skip this example in windows 🤔

**Comment by uilianries** [comment]
Did you try the make installer from CCI?

**Comment by danimtb** [comment]
It could be an alternative but I did not want to add more noise to the example. Will think about it

**Comment by uilianries** [comment]
Indeed, but usually people don't have make.exe installed on Windows.

**Comment by danimtb** [comment]
Needs #60

File: features/makefiles/hellolib/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+class HelloConan(ConanFile):
+    name = "hello"
+    version = "0.1"
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake"
+    exports_sources = "src/*"
+
+    def build_requirements(self):
+        if self.settings.os == "Windows":
+            self.build_requires("make/4.2.1")
```
Comment (danimtb) [inline features/makefiles/hellolib/conanfile.py:14]: This was added to make the example build on windows but maybe is too complex for the example in the docs

File: features/makefiles/helloapp/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+class AppConan(ConanFile):
+    name = "app"
+    version = "0.1"
+    settings = "os", "compiler", "build_type", "arch"
+    exports_sources = "src/*"
+    requires = "hello/0.1@user/testing"
+
+    def build_requirements(self):
+        if self.settings.os == "Windows":
+            self.build_requires("make/4.2.1")
```
Comment (danimtb) [inline features/makefiles/helloapp/conanfile.py:14]: This was added to make the example build on windows but maybe is too complex for the example in the docs

Comment (danimtb) [reply features/makefiles/helloapp/conanfile.py:14]: Should we skip this example on windows and remove this?

File: features/makefiles/helloapp/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+    def build(self):
+        with tools.chdir("src"):
+            env_build = AutoToolsBuildEnvironment(self)
+            env_build.make()
```
Comment (danimtb) [inline features/makefiles/helloapp/conanfile.py:19]:
```suggestion
            atools = AutoToolsBuildEnvironment(self)
            atools.make()
```

File: features/makefiles/hellolib/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+    def build(self):
+        with tools.chdir("src"):
+            env_build = AutoToolsBuildEnvironment(self)
+            env_build.make()
```
Comment (danimtb) [inline features/makefiles/hellolib/conanfile.py:19]:
```suggestion
            atools = AutoToolsBuildEnvironment(self)
            atools.make()
```

File: features/makefiles/helloapp/conanfile.py (hunk 2 above)
Comment (memsharded) [reply features/makefiles/helloapp/conanfile.py:14]: Probably yes, this example can assume that you have Make in the system. Better to keep examples as lean as possible.

Comment (memsharded) [reply features/makefiles/helloapp/conanfile.py:14]: But make sure to put a big comment with a warning about this.

Comment (uilianries) [reply features/makefiles/helloapp/conanfile.py:14]: I think so. As examples are supposed to be used by newbies, I believe only Linux users will run make.

**Pull request merged**